runner/queries.go

```diff
@@ -47,21 +47,31 @@ func gatherQueryStatsForServer(ctx context.Context, server *state.Server, global
 	if err != nil {
 		return newState, errors.Wrap(err, "error collecting pg_stat_statements")
 	}
+	_, newState.PlanStats, err = postgres.GetPlans(ctx, server, logger, connection, globalCollectionOpts, postgresVersion)
+	if err != nil {
+		return newState, errors.Wrap(err, "error collecting query plan stats")
+	}
 
 	// Don't calculate any diffs on the first run (but still update the state)
 	if len(server.PrevState.StatementStats) == 0 || server.PrevState.LastStatementStatsAt.IsZero() {
 		return newState, nil
 	}
 
 	diffedStatementStats := diffStatements(newState.StatementStats, server.PrevState.StatementStats)
+	diffedPlanStats := diffPlanStats(newState.PlanStats, server.PrevState.PlanStats)
 	collectedIntervalSecs := uint32(newState.LastStatementStatsAt.Sub(server.PrevState.LastStatementStatsAt) / time.Second)
 
 	timeKey := state.PostgresStatementStatsTimeKey{CollectedAt: collectedAt, CollectedIntervalSecs: collectedIntervalSecs}
 	newState.UnidentifiedStatementStats = server.PrevState.UnidentifiedStatementStats
+	newState.UnidentifiedPlanStats = server.PrevState.UnidentifiedPlanStats
 	if newState.UnidentifiedStatementStats == nil {
 		newState.UnidentifiedStatementStats = make(state.HistoricStatementStatsMap)
 	}
+	if newState.UnidentifiedPlanStats == nil {
+		newState.UnidentifiedPlanStats = make(state.HistoricPlanStatsMap)
+	}
 	newState.UnidentifiedStatementStats[timeKey] = diffedStatementStats
+	newState.UnidentifiedPlanStats[timeKey] = diffedPlanStats
```

Comment:
I feel like interleaving these two code paths is going to make it harder to maintain going forward. It'd be better to put all of the new code after the existing code.